internal/bminventory/inventory_v2_handlers.go

```diff
@@ -672,6 +673,28 @@ func (b *bareMetalInventory) canServeHostIPXEScript(infraEnv *common.InfraEnv, m
 	return nil
 }
 
+func kernelArgsToSlice(infraEnv *common.InfraEnv) (models.KernelArguments, error) {
+	if infraEnv.DiscoveryKernelArguments == nil {
+		return nil, nil
+	}
+	var ret models.KernelArguments
+	if err := json.Unmarshal([]byte(swag.StringValue(infraEnv.DiscoveryKernelArguments)), &ret); err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+func kernelArgsAppendStr(infraEnv *common.InfraEnv) (string, error) {
+	kernelArguments, err := kernelArgsToSlice(infraEnv)
+	if err != nil {
+		return "", err
+	}
+	if len(kernelArguments) > 0 {
+		return " " + strings.Join(kernelArguments, " "), nil
```

Comment:
why do you need the space at the start?

Reply:
To separate from the existing line

Reply:
then why not have the space there? it looks strange to have it in this function, or at least not clear why it's needed

Reply:
Because I didn't want to add an extra space if no parameter follows it.

Reply:
This test https://github.com/openshift/assisted-service/blob/938229b5f6f77c3d0331c18324e2e73534ff9604/internal/bminventory/inventory_test.go#L9325 will probably fail if it will be added unconditionally.